modules/axhal/src/dma.rs

```diff
@@ -0,0 +1,26 @@
+use dma_api::Impl;
```

Comment:
I think the `Impl` trait exposed by `dma-api` crate is strange.

I believe that you want to use this trait for the translation from `IOVA` to actual `PA`

Reply:
maybe you can refer to the DMA addr related trait defined in `virtio-drivers`? 

https://github.com/rcore-os/virtio-drivers/blob/master/src/hal.rs

Reply:
Sorry, I need to learn how `dma_api` works before offering any more suggestions.

kind of busy lately, so this may take a little bit longer